x-pack/legacy/plugins/apm/public/hooks/useFetcher.tsx

```diff
@@ -18,42 +18,39 @@ export enum FETCH_STATUS {
   FAILURE = 'failure'
 }
 
-type Fetcher = (...args: any[]) => any;
-type GetReturn<TFetcher extends Fetcher> = Exclude<
-  ReturnType<TFetcher>,
-  undefined
-> extends Promise<infer TReturn>
-  ? TReturn
-  : ReturnType<TFetcher>;
```

Comment:
@dgieselaar fyi what we talked about earlier

Reply:
As far as I can tell, using `prevResult` results in both the return type of the fetcher function and the `prevResult` argument being `any`. [Here's a simplified demo](https://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABMOdEApEEpEG8BQ+ixy408SIAzgKYBiNUEAFjQE4A8AKgCICGUPgD4MwMAC5MABzY0AbgCUaVEABsoAfnG8BfbIgC8InYJx4iJS7Kgg2SApcck5fVSBrjkYdLgC+iPioAsABPHEDEEz4LJ18AbhjEX0JLBydEAHoMxBc3GkNgkIAaRBl5JRV1Ar5QxMsIBCooPBzXdySC6npGFnZ0aVlFZTVmnCNzdMdrW3tkVE8AcgWkhMnfLFWSZMS0pyzWvILcObhPJrYYMABzJJKyocrmgxaUU8Rzy5vkyYawJpbcu1nsdXotlv5-M8ugwmKw2P1SoMKiN9ONdpNENM7C95oglis6lsNoltqlCcR9oD8sCTmcoBdrklyYhfv9jlSOlDaDDevDMGMROjJljZqC8eDNrFiZZviRCMkgA).

The only reason it "works" is because TypeScript uses the type of the default parameters.

Reply:
btw, what happens when you use `data` instead of `prevResult`?